### PR TITLE
Add section on handling of invalid data, invalidate excess ...

### LINKED ISSUE #87: Missing specification of wasted bits-per-sample value
It is not immediately obvious howwhen to apply the wasted bits-per-sample value during decoding. It's possible to guess that a left shift involved, but it is unclear whether this happens after or before LPC decoding. (The impact on the bits-per-sample value of the subframe is also unspecified, see #83.)

### PR DESCRIPTION
**Pull request by ktmf01**
... wasted bits

This fixes #87

### CODE REVIEW

File: flac.md [hunk 1]
```diff
@@ -178,6 +179,13 @@ There are several ways to convert unsigned sample representations to signed samp
 
 Unary coding in a FLAC bitstream is done with zero bits terminated with a one bit, e.g. the number 5 is coded unary as 0b000001. This prevents the frame sync code from appearing in unary coded numbers.
 
+When a FLAC file contains invalid data, decoder behavior is left unspecified. A decoder MAY choose to stop decoding on encountering such invalid data. Examples of such invalid data are
+
+- One or more decoded sample values exceeds the range offered by the bit depth as coded for that frame. E.g. in a frame with a bit depth of 8 bits, any samples not in the inclusive range from -128 to 127 are invalid.
+- The number of wasted bits (see section [wasted bits per sample](wasted-bits-per-sample)) of a subframe is such that the bit depth of that subframe (see section [constant subframe](#constant-subframe) for a description of subframe bit depth) equals zero or is negative
```
Comment (kieranjol) [inline flac.md:185]: Should it be #wasted-bits-per-sample?

File: flac.md [hunk 2]
```diff
@@ -178,6 +179,13 @@ There are several ways to convert unsigned sample representations to signed samp
...
+- A frame header CRC (see section [frame header CRC](#frame-header-crc)) or frame footer CRC (see section [frame footer](#frame-footer)) does not validate
+- One of the invalid bit patterns described in table (#tableinvalidpatterns, use counter) above is used
```
Comment (kieranjol) [inline flac.md:187]: The link to tableinvalidpattern might be broken?

File: flac.md (hunk 1 above)
Comment (ktmf01) [reply flac.md:185]: Yes, thanks

File: flac.md (hunk 2 above)
Comment (ktmf01) [reply flac.md:187]: That is a special syntax for mmark to embed a table number. Works fine on my end, but it needs a recent version of mmark and xml2rfc to work.

**Review by retokromer — APPROVED** [review]

File: flac.md (hunk 2 above)
Comment (kieranjol) [reply flac.md:187]: Cool!

**Pull request merged**